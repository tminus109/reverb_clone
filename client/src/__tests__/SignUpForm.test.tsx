import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { cleanup, render, screen } from "@testing-library/react";
import SignUpForm from "../components/forms/SignUpForm";

afterEach(cleanup);

test("Form renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SignUpForm />, div);
});

test("Snapshot", () => {
  const tree = renderer.create(<SignUpForm />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("firstName-input tests", () => {
  beforeEach(() => {
    render(<SignUpForm />);
  });

  test("Expect firstName input to be in the document", async () => {
    const firstNameInput = await screen.findByTestId("first_name");
    expect(firstNameInput).toBeInTheDocument();
  });
});
