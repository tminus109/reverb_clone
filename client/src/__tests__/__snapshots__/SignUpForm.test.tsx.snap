// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
<div
  className="signup_form"
>
  <h1>
    Create Your Account
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="first_name"
    >
      First Name:
      <input
        data-testid="first_name"
        id="first_name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="last_name"
    >
      Last Name:
      <input
        data-testid="last_name"
        id="last_name"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="email"
    >
      Email:
      <input
        data-testid="email"
        id="email"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      
    </label>
    <label
      htmlFor="email2"
    >
      Email Confirmation:
      <input
        data-testid="email2"
        id="email2"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      
    </label>
    <label
      htmlFor="password"
    >
      Password:
      <input
        data-testid="password"
        id="password"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
      
    </label>
    <span>
      (at least 8 characters)
    </span>
    <input
      type="submit"
    />
    <div
      className="message"
    >
      
    </div>
  </form>
</div>
`;
